addons/sale/models/sale_order.py

```diff
@@ -908,22 +917,50 @@ def _action_confirm(self):
                     order._create_analytic_account()
 
     def _send_order_confirmation_mail(self):
-        if not self:
+        """ Send a mail to the SO customer to inform them that their order has been confirmed.
+
+        :return: None
+        """
+        if not self:  # This method can be called with an empty recordset and
+                      # calling `_get_confirmation_template` will crash.
+            return
+        mail_template = self._get_confirmation_template()
```

Comment:
Wrong, `_get_confirmation_template` is `ensure_one()`, it must stay in the loop here.